akka-persistence-typed/src/test/java/akka/persistence/typed/javadsl/PersistentActorJavaDslTest.java

```diff
@@ -559,4 +565,68 @@ public Incremented fromJournal(Wrapper wrapper) {
   }
   // event-wrapper
 
+  static class IncorrectExpectedStateForThenRun
+      extends EventSourcedBehavior<String, String, Object> {
+
+    private final ActorRef<String> startedProbe;
+
+    public IncorrectExpectedStateForThenRun(
+        ActorRef<String> startedProbe, PersistenceId persistenceId) {
+      super(persistenceId);
+      this.startedProbe = startedProbe;
+    }
+
+    @Override
+    public Object emptyState() {
+      return 1;
+    }
+
+    @Override
+    public CommandHandler<String, String, Object> commandHandler() {
+      return newCommandHandlerBuilder()
+          .forAnyState()
+          .matchCommand(
+              msg -> msg.equals("expect wrong type"),
+              (context) ->
+                  Effect()
+                      .none()
+                      .thenRun(
+                          (String wrongType) -> {
+                            // wont happen
+                          }))
+          .build();
+    }
+
+    @Override
+    public void onRecoveryCompleted(Object o) {
+      startedProbe.tell("started!");
+    }
+
+    @Override
+    public EventHandler<Object, String> eventHandler() {
+      return newEventHandlerBuilder()
+          .forAnyState()
+          .matchAny((event, state) -> state); // keep Integer state
+    }
+  }
+
+  @Test
+  public void failOnIncorrectExpectedStateForThenRun() {
+    TestProbe<String> probe = testKit.createTestProbe();
+    ActorRef<String> c =
+        testKit.spawn(
+            new IncorrectExpectedStateForThenRun(probe.getRef(), new PersistenceId("foiesftr")));
+
+    probe.expectMessage("started!"); // workaround for #26256
+
+    new EventFilter(Logging.Error.class, Adapter.toUntyped(testKit.system()))
```

Comment:
Maybe a more specific filter as there are some other error logs from this class:

```
[ERROR] [02/04/2019 09:55:36.435] [PersistentActorJavaDslTest-akka.actor.default-dispatcher-9] [akka://PersistentActorJavaDslTest/user] akka.persistence.RecoveryPermitter$RecoveryPermitGranted$ cannot be cast to akka.persistence.typed.javadsl.PersistentActorJavaDslTest$Command
java.lang.ClassCastException: akka.persistence.RecoveryPermitter$RecoveryPermitGranted$ cannot be cast to akka.persistence.typed.javadsl.PersistentActorJavaDslTest$Command
	at akka.persistence.typed.javadsl.PersistentActorJavaDslTest$6.aroundReceive(PersistentActorJavaDslTest.java:466)
	at akka.actor.typed.internal.InterceptorImpl.receive(InterceptorImpl.scala:73)
	at akka.actor.typed.Behavior$.interpret(Behavior.scala:421)
	at akka.actor.typed.Behavior$.interpretMessage(Behavior.scala:394)
	at akka.actor.typed.internal.adapter.ActorAdapter.akka$actor$typed$internal$adapter$ActorAdapter$$handleMessage(ActorAdapter.scala:82)
	at akka.actor.typed.internal.adapter.ActorAdapter$$anonfun$running$1.applyOrElse(ActorAdapter.scala:78)
	at akka.actor.Actor.aroundReceive(Actor.scala:517)
	at akka.actor.Actor.aroundReceive$(Actor.scala:515)
	at akka.actor.typed.internal.adapter.ActorAdapter.aroundReceive(ActorAdapter.scala:39)
	at akka.actor.ActorCell.receiveMessage(ActorCell.scala:592)
	at akka.actor.ActorCell.invoke(ActorCell.scala:561)
	at akka.dispatch.Mailbox.processMailbox(Mailbox.scala:258)
	at akka.dispatch.Mailbox.run(Mailbox.scala:225)
	at akka.dispatch.Mailbox.exec(Mailbox.scala:235)
	at akka.dispatch.forkjoin.ForkJoinTask.doExec(ForkJoinTask.java:260)
	at akka.dispatch.forkjoin.ForkJoinPool$WorkQueue.runTask(ForkJoinPool.java:1339)
	at akka.dispatch.forkjoin.ForkJoinPool.runWorker(ForkJoinPool.java:1979)
	at akka.dispatch.forkjoin.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:107)

```